ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -142,14 +142,15 @@ export default class PermissionConnect extends Component {
     }
   }
 
-  _doRedirect () {
-    const { history, hasPendingPermissionsRequests } = this.props
+  _doRedirect = async () => {
+    const { history, hasPendingPermissionsRequests, rejectPermissionsRequest, permissionsRequestId } = this.props
 
     if (
       !hasPendingPermissionsRequests &&
       getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION
     ) {
       global.platform.closeCurrentWindow()
+      await rejectPermissionsRequest(permissionsRequestId)
```

Comment:
It seems that this situation can only happen if `permissionsRequest` is falsy (see `componentDidUpdate`). If this request is still pending, then why is the `permissionsRequest` prop falsy? Something doesn't seem right here...